osf/models/analytics.py

```diff
@@ -124,6 +124,11 @@ def update_counter(cls, page, node_info):
                     model_instance.save()
                     return
 
+            if page_type == 'view' and node_info:
```

Comment:
As discussed: For the else blocks here, send to sentry the context for all of the variables that go into the decision, so we can see why it's not going through.

Not discussed: maybe also log if this succeeds, so we can see if maybe just saving isn't doing enough for this.

Reply:
The page_type is the riskiest part of this (although it should be fine because we've used it, there's a lot of code that goes into it and I don't think it's explicitly tested. Before this is done, we may pull out the cleaning and page type determination into its own function so that we can unit test those under different circumstances, but if we get this problem working, that will be a follow-up ticket.

Reply:
Also, if anything fails in https://github.com/CenterForOpenScience/osf.io/pull/7693/files#diff-4519792f13c44cf51a8eec0288fa674aR81 then everything's going to fail, as it's wrapped in an atomic transaction.